fathom/src/source.rs

```diff
@@ -77,7 +80,9 @@ impl StringInterner {
 }
 
 /// File id.
-pub type FileId = usize; // TODO: use wrapper struct
+// 4 billion files should be enough for anyone, and `u16` doesn't save any size in `ByteRange` or `Span`.
+// `NonZeroU32` saves 4 bytes on the size of `Span` compared to `u32`.
+pub type FileId = NonZeroU32; // TODO: use wrapper struct
```

Comment:
Nice! I like the use of a `NonZeroU32`. I think it’d be nice to use a wrapper struct here to hide the internal representation.